client/starwhale/core/dataset/model.py

```diff
@@ -121,6 +121,7 @@ def __init__(self, uri: URI) -> None:
             str, t.Any
         ] = {}  # TODO: use manifest class get_conda_env
         self.yaml_name = DefaultYAMLName.DATASET
+        self._version = uri.object.version if uri.object and uri.object.version else ""
```

Comment:
Why do we set the `self._version` in the init process?
The current version logic is in the [ _gen_version](https://github.com/star-whale/starwhale/blob/main/client/starwhale/base/bundle.py#L128) . All bundles(model/runtime/dataset) follow the unify gen version process.

Reply:
this is the same as https://github.com/star-whale/starwhale/pull/1552#discussion_r1033057474